cpp/src/gandiva/tests/generate_data.h

```diff
@@ -79,6 +81,24 @@ class Int64DataGenerator : public DataGenerator<int64_t> {
   Random random_;
 };
 
+class Decimal128DataGenerator : public DataGenerator<arrow::Decimal128> {
+ public:
+  explicit Decimal128DataGenerator(bool large) : large_(large) {}
+
+  arrow::Decimal128 GenerateData() {
+    uint64_t low = random_.next();
+    int64_t high = random_.next();
+    if (large_) {
+      high += (1ull << 62);
+    }
+    return arrow::Decimal128(high, low);
+  }
+
+ protected:
+  bool large_;
+  Random random_;
+};
```

Comment:
Let us try to prioritize consolidating our random data generation code

Reply:
will do